src/Components/DashboardComponents/Profile.js

```diff
@@ -1,309 +1,345 @@
-    }
-
-    const cancel = () => {
-        setEditedUserDetails(userDetails)
-    }
-
-    const photoChangeHandler = (e) => {
-        setPhoto(e.target.files[0]);
-    }
+  .bottom-container {
+    width: 90%;
+    display: flex;
+    justify-content: flex-end;
+    margin-top: 3rem;
 
-    const uploadProfilePhoto = () => {
-        setWaiting(true);
-        const { _id } = loggedInUser;
-        const formData = new FormData();
-        formData.append(
-            'image',
-            photo
-        )
-
-        axiosWithBase.post(`/upload/${_id}`, formData)
-            .then(() => {
-                populateUserDetails();
-                setWaiting(false);
-            })
-            .catch(() => {
-                setWaiting(false);
-                toast.error("Error uploading photo")
-            })
+    #cancel {
+      width: 106px;
+      height: 36px;
+      background: #ffa439;
+      border: none;
+      color: white;
+      border-radius: 4px;
+      cursor: pointer;
+      transition: opacity 0.2s ease-in-out;
+      font-size: 1.2rem;
+
+      &:focus {
+        outline: none;
+      }
+
+      &:hover {
+        opacity: 0.5;
+        transition: opacity 0.2s ease-in-out;
+      }
     }
 
-    const populateUserDetails = () => {
-        const { _id } = loggedInUser;
-        const token = localStorage.getItem('authorization');
-
-        axiosWithBase.get(`/profile/${_id}`, { headers: { 'authorization': token } })
-            .then((res) => {
-                setUserDetails(res.data)
-                setEditedUserDetails(res.data)
-                setWaiting(false)
-            })
-            .catch(() => {
-                toast.error("There was an error retrieving your information.")
-            })
+    #save {
+      width: 106px;
+      height: 36px;
+      background: rgba(49, 215, 21, 0.8);
+      color: white;
+      border: none;
+      margin-left: 1rem;
+      cursor: pointer;
+      border-radius: 4px;
+      transition: opacity 0.2s ease-in-out;
+      font-size: 1.2rem;
+
+      &:focus {
+        outline: none;
+      }
+
+      &:hover {
+        opacity: 0.5;
+        transition: opacity 0.2s ease-in-out;
+      }
     }
-
-    useEffect(() => {   
+  }
+`;
+
+function Profile({ loggedInUser, setUserDetails, userDetails, ...props }) {
+  const [editedUserDetails, setEditedUserDetails] = useState();
+  const [waiting, setWaiting] = useState(true);
+  const [photo, setPhoto] = useState(null);
+  const urlString = queryString.parse(props.location.search);
+
+  function googleUserDetails(urlString) {
+    const urlToken = urlString.token;
+    const decodedUrlToken = jwt.decode(urlToken);
+    const google_id = decodedUrlToken.subject;
+    localStorage.setItem("authorization", urlToken);
+    return google_id;
+  }
+
+  let _id = null;
+  Object.keys(urlString).length
+    ? (_id = googleUserDetails(urlString))
+    : (_id = loggedInUser.userId);
```

Comment:
This is the point that breaks the application if a user logs in with a normal account...there is no userID on the loggedInUser....it should be `loggedInUser._id`